research/berry/berrylib/fast_inla.py

```diff
@@ -328,59 +337,94 @@ def step(args):
     # )
     n_arms = y.shape[0]
     theta_max0 = jnp.zeros(n_arms)
+    init_args = (0, theta_max0, jnp.zeros((n_arms, n_arms)), True)
 
-    out = jax.lax.while_loop(
-        lambda args: args[2], step, (theta_max0, jnp.zeros((n_arms, n_arms)), True)
-    )
-    theta_max, hess_inv, stop = out
+    if fast_loop:
+        out = jax.lax.while_loop(
+            lambda args: ((args[0] < max_iter) & args[-1]), step, init_args
+        )
+    else:
+        args = init_args
+        step = jax.jit(step)
+        for i in range(max_iter):
+            args = step(args)
+            out = args
+            if not args[-1]:
+                break
+    i, theta_max, hess_inv, go = out
     return theta_max, hess_inv
```

Comment:
Unless the performance cost is large, I think we should return `i` here too. It'll provide a nice degree of visibility into the optimizer and let us confirm that it converged from outside the jit jax code.